matching-notification-resources/src/main/resources/resources/uicomponents/matchingNotification/matchesTable.js

```diff
@@ -1614,57 +1631,41 @@ var PhenoTips = (function (PhenoTips) {
         }.bind(this));
 
         matchesToSet.each( function(match, index) {
+            var matchIndex = this._matches.indexOf(match);
+            var cachedIndex = this._cachedMatches.indexOf(match);
+
             if (properties.hasOwnProperty('notified')) {
-                // update macthes status
-                if (notifiedPatients && notifiedPatients.hasOwnProperty('isAdminNotification')) {
-                    this._matches[this._matches.indexOf(match)].adminNotified = true;
-                    this._cachedMatches[this._cachedMatches.indexOf(match)].adminNotified = true;
-                } else {
-                    this._matches[this._matches.indexOf(match)].contacted = true;
-                    this._cachedMatches[this._cachedMatches.indexOf(match)].contacted = true;
-                }
-                // updating contact button
-                var contactTd = this._tableElement.down('tr[data-matchid="' + match.id +'"] td[name="contact"]');
-                if (contactTd && contactTd.down('.contact-button')) {
-                    if (properties.notified) {
-                        contactTd.down('.contact-button').addClassName('secondary');
-                    } else {
-                        contactTd.down('.contact-button').removeClassName('secondary');
-                    }
+                if (properties.successNotificationHistories && Object.keys(properties.successNotificationHistories).indexOf(match.id.toString()) > -1) {
+                    this._matches[matchIndex].notificationHistory = JSON.stringify(properties.successNotificationHistories[match.id]);
+                    this._matches[matchIndex].notificationHistory = this._organiseNotificationHistory(this._matches[matchIndex]);
```

Comment:
This is a very weird pattern `z.h = data; z.h = f(z)`. A separate issue PN-438 to fix this was created